sympy/matrices/expressions/matpow.py

```diff
@@ -100,7 +99,15 @@ def _eval_conjugate(self):
         return MatPow(base.conjugate(), exp)
 
     def _eval_derivative(self, x):
-        return Pow._eval_derivative(self, x)
+        assert not hasattr(x, "shape")
+        # dbase = self.base.diff(x)
+        dexp = self.exp.diff(x)
+        if dexp != 0:
+            raise NotImplementedError("cannot derive matrix power if exponent depends on the derivation variable")
+        if self.exp.is_Integer:
+            from sympy import MatMul
+            return MatMul.fromiter(self.base for i in range(self.exp))._eval_derivative(x)
+        raise NotImplementedError("cannot evaluate derivatives of matrix power is not a known integer")
```

Comment:
This will fail if exp is a negative integer. Also I think i is beter o have unevaluated derivatives rather than raising an exceptions. The alternative conditions here do not mean that the derivatives is undefined or anything e.g.:
```
In [8]: M = MatPow(Matrix([[0, m], [n, 0]]), x)

In [9]: M
Out[9]:
      x
⎡0  m⎤
⎢    ⎥
⎣n  0⎦

In [10]: M.diff(m)
---------------------------------------------------------------------------
NotImplementedError
```

Reply:
should be fixed now